Wrappers/Python/cil/utilities/quality_measures.py

```diff
@@ -18,36 +18,90 @@
 
 from cil.optimisation.functions import L2NormSquared, L1Norm
+from cil.framework import DataContainer
 import numpy as np
 
 
-def mse(dc1, dc2):    
+
+def mse(dc1, dc2, mask=None):    
     
     ''' Returns the Mean Squared error of two DataContainers
-    '''
     
-    diff = dc1 - dc2    
-    return L2NormSquared().__call__(diff)/dc1.size
+    Parameters
+    ----------
+    dc1: `DataContainer`
+        One image to be compared
+    dc2: `DataContainer`
+        Second image to be compared 
+    mask: array or `DataContainer` of Boolean values or 0's and 1's with the same dimensions as the `dc1` and `dc2`
+        Region of interest for the calculation. 
+    '''  
+
+    if mask is None:
+        diff = dc1 - dc2    
+        return L2NormSquared().__call__(diff)/dc1.size
+    else:
+        if isinstance(mask, DataContainer):
+            mask = mask.as_array()
+        return np.mean(((dc1.as_array() - dc2.as_array())**2), where=mask.astype('bool'))
 
 
-def mae(dc1, dc2):
+def mae(dc1, dc2, mask=None):
     
     ''' Returns the Mean Absolute error of two DataContainers
-    '''    
     
-    diff = dc1 - dc2  
-    return L1Norm().__call__(diff)/dc1.size
+        
+    Parameters
+    ----------
+    dc1: `DataContainer`
+        One image to be compared
+    dc2: `DataContainer`
+        Second image to be compared 
+    mask: array or `DataContainer` of Boolean values or 0's and 1's with the same dimensions as the `dc1` and `dc2`
+        Region of interest for the calculation. 
+        
+        
+    '''    
+    if mask is None:
+        diff = dc1 - dc2  
+        return L1Norm().__call__(diff)/dc1.size
+    else:
+        if isinstance(mask, DataContainer):
+            mask=mask.as_array()
+    return np.mean(np.abs((dc1.as_array()-dc2.as_array())), where=mask.astype('bool'))
 
-def psnr(ground_truth, corrupted, data_range = 255):
+def psnr(ground_truth, corrupted, data_range=None, mask=None):
 
     ''' Returns the Peak signal to noise ratio
-    '''   
     
-    tmp_mse = mse(ground_truth, corrupted)
+    Parameters
+    ----------
+    ground_truth: `DataContainer`
+        The reference image
+    corrupted: `DataContainer`
+        The image to be evaluated 
+    data_range: scalar value, default=None
+        PSNR scaling factor, the dynamic range of the images (i.e., the difference between the maximum the and minimum allowed values). To match with scikit-image the default is ground_truth.array.max()
+    mask: array or `DataContainer` of Boolean values or 0's and 1's with the same dimensions as the `ground_truth` and `corrupted`
+        Region of interest for the calculation. 
+    '''  
+    if data_range is None:
+        if mask is None:
+            data_range=ground_truth.array.max()
+        else:
+            if isinstance(mask, DataContainer):
+                mask=mask.as_array()
+            data_range=np.amax(ground_truth.as_array()[mask.astype('bool')])
```

Comment:
is this different to `np.max`. `np.max` also takes `where` as an argument so you can pass the mask directly.